assets/src/modules/Snapping.js

```diff
@@ -0,0 +1,190 @@
+import { mainLizmap, mainEventDispatcher } from '../modules/Globals.js';
+
+export default class Snapping {
+
+    constructor() {
+
+        this._active = false;
+        this._snapLayersRefreshable = false;
+
+        this._maxFeatures = 1000;
+        this._restrictToMapExtent = true;
+        this._config = undefined;
+
+        // Create layer to store snap features
+        const snapLayer = new OpenLayers.Layer.Vector('snaplayer', {
+            visibility: false,
+            styleMap: new OpenLayers.StyleMap({
+                pointRadius: 2,
+                fill: false,
+                stroke: true,
+                strokeWidth: 3,
+                strokeColor: 'red',
+                strokeOpacity: 0.8
+            })
+        });
+
+        mainLizmap.lizmap3.map.addLayer(snapLayer);
+
+        const snapControl = new OpenLayers.Control.Snapping({
+            layer: mainLizmap.edition.editLayer,
+            targets: [{
+                layer: snapLayer
+            }]
+        });
+        mainLizmap.lizmap3.map.addControls([snapControl]);
+        mainLizmap.lizmap3.controls['snapControl'] = snapControl;
+
+        this._setSnapLayersRefreshable = () => {
+            if(this._active){
+                this.snapLayersRefreshable = true;
+            }
+        }
+
+        // Activate snap when a layer is edited
+        mainEventDispatcher.addListener(
+            () => {
+                // Get snapping configuration for edited layer
+                for (const editionLayer in mainLizmap.config.editionLayers) {
+                    if (mainLizmap.config.editionLayers.hasOwnProperty(editionLayer)) {
+                        if (mainLizmap.config.editionLayers[editionLayer].layerId === mainLizmap.edition.layerId){
+                            const editionLayerConfig = mainLizmap.config.editionLayers[editionLayer];
+                            if (editionLayerConfig.hasOwnProperty('snap_layers')){
+                                this.config = {
+                                    'snap_layers': editionLayerConfig.snap_layers.split(','),
+                                    'snap_vertices': (editionLayerConfig.hasOwnProperty('snap_vertices') && editionLayerConfig.snap_vertices === 'True') ? true : false,
+                                    'snap_segments': (editionLayerConfig.hasOwnProperty('snap_segments') && editionLayerConfig.snap_segments === 'True') ? true : false,
+                                    'snap_intersections': (editionLayerConfig.hasOwnProperty('snap_intersections') && editionLayerConfig.snap_intersections === 'True') ? true : false,
```

Comment:
```
editionLayerConfig.snap_intersections === 'True') ? true : false,
```
Side note, I think there should be a function to parse a boolean value `"True"`, true bool value `True` (`"true"`, `1` too?).
I hope for Lizmap 4, I will close your issue : https://github.com/3liz/lizmap-plugin/issues/176. It's a single line to uncomment in the plugin and it will change everywhere in the CFG.
So you might not want to update all parts in Lizmap where you are reading boolean values.

Reply:
I will make changes in all JavaScript for Lizmap 4.